pages/my-checkins.tsx

```diff
@@ -21,14 +22,33 @@ export default function MyCheckinsPage() {
   const [checkoutFn, { error }] = useMutation(checkout)
 
   const handleCheckout = React.useCallback(
-    async (checkin) => {
+    async (checkins: Checkin[]) => {
       setIsLoading(true)
-      await checkoutFn({ id: checkin.id })
+      await Promise.all(
+        checkins.map((checkin) => checkoutFn({ id: checkin.id }))
+      )
       setIsLoading(false)
     },
     [setIsLoading, checkoutFn]
   )
 
+  // Sorts checkins by time and groups proxy checkins together with their "main" checkins
+  const groupedCheckins = React.useMemo(() => {
+    const sortedCheckins = checkinsInfo.data?.sort(
+      (c1, c2) => c1.enteredAt.getTime() - c2.enteredAt.getTime()
+    )
+
+    return sortedCheckins?.reduce((result: Checkin[][], checkin: Checkin) => {
```

Comment:
Can you maybe add a small comment what's happening here? Took me a while to understand how  you're grouping them

Reply:
Sure :+1: